pkg/apis/wsecurity/v1/count.go

```diff
@@ -0,0 +1,146 @@
+package v1
+
+import (
+	"bytes"
+	"fmt"
+)
+
+//////////////////// CountProfile ////////////////
+
+// Exposes ValueProfile interface
+type CountProfile uint8
+
+func (profile *CountProfile) Profile(args ...interface{}) {
+	*profile = CountProfile(args[0].(uint8))
+}
+
+func (profile *CountProfile) String(depth int) string {
+	return fmt.Sprintf("%d", uint8(*profile))
+}
+
+//////////////////// CountPile ////////////////
+
+// Exposes ValuePile interface
+type CountPile []uint8
```

Comment:
xxxPile Data types will pile the dataType, but it will not always be as simple as an array
We should really keep each of the data types 
- xxxProfile
- xxxPile
- xxxConfig   
as a standalone and not derived from another

Reply:
Do you want to use Go 1.18 generics (rather than using concrete types here?

e.g.
```golang
type Profile[T any] struct {
  T value
}

func (p *Profile[T]) Profile(args ...interface{}) {
  // ...
}

// OR

func (p *Profile[T]) Profile(args ...T) {
  // ...
}
```

Reply:
This is an interesting discussion to have.

When can generics help?
Generics can help when we have multiple data types that can share the **same** implementation. 
SG uses multiple data types (more than 14 right now) that each requires its own very different implementation... but they are all sharing the same basic interface.

E.g. `XXXConfig.Decide()` if very different in its implementation from `YYYConfig.Decide()`. The same is true for other methods of the XXXConfig/Pile/Profile and YYYConfig/Pile/Profile data types.

So using generics may not offer us any immediate code reduction, but can help make some of the more basic data types (such as Count here) more generic :). E.g. Count can be made to support uint8, unit16, and uint32 counters in one stroke. 

For example:
```golang
type CountPile[T any] []T

func (pile *CountPile[T]) Merge(otherValPile ValuePile) {
	otherPile := otherValPile.(*CountPile[T])
	*pile = append(*pile, *otherPile...)
}

var a,b CountPile[uint8]
...
a.Merge(&b)
```

---
SG uses Kubernetes gengo, right now it does not seem to support generics code. 
We can add this as we move forward.

Reply:
> SG uses multiple data types (more than 14 right now) that each requires its own very different implementation

Just thinking out loud, but would it be possible to come up with a more generic data model that can better accommodate those different existing types? (Good chance there's not, but this might be the easiest time to revisit the data model if there is, so figured it was worth asking)

Reply:
We can revisit this question at the next PR when we get to see 3 more data types: 
- SimpleVal which uses Count (shown here), AsciiFlags, and FlagsSlice

Reply:
Does it make sense to include these to make it more clear what varies?

The simplicity of the existing operations makes it hard to understand the end-goal state.

Reply:
sure, expect a new PR soon with these as well later today